File: scripts/remove_csv_duplicates.js
#!/usr/bin/env node
"use strict";

/**
 * Remove duplicate matches from CSV
 * Keeps first occurrence of each match ID
 */

const fs = require('fs');
const path = require('path');

const inputFile = process.argv[2] || path.join(__dirname, '../out/matches_detailed.csv');
const outputFile = process.argv[3] || inputFile.replace('.csv', '_unique.csv');

console.log('📥 Reading CSV:', inputFile);

const csvContent = fs.readFileSync(inputFile, 'utf8');
const lines = csvContent.split('\n');

const header = lines[0];
const dataLines = lines.slice(1).filter(line => line.trim());

console.log('📊 Original rows:', dataLines.length);

// Track seen match IDs
const seen = new Set();
const uniqueLines = [];

for (const line of dataLines) {
  const matchId = line.split(',')[0];
  
  if (!seen.has(matchId)) {
    seen.add(matchId);
    uniqueLines.push(line);
  }
}

console.log('✅ Unique rows:', uniqueLines.length);
console.log('❌ Duplicates removed:', dataLines.length - uniqueLines.length);

// Write cleaned CSV
const cleaned = [header, ...uniqueLines].join('\n');
fs.writeFileSync(outputFile, cleaned);

console.log('💾 Saved to:', outputFile);

File: scripts/remove_csv_duplicates.py
#!/usr/bin/env python3
"""
Remove duplicate matches from CSV
Keeps first occurrence of each match ID
"""

import sys
import os

def remove_duplicates(input_file, output_file=None):
    if not output_file:
        base, ext = os.path.splitext(input_file)
        output_file = f"{base}_unique{ext}"
    
    print(f'📥 Reading CSV: {input_file}')
    
    with open(input_file, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    header = lines[0]
    data_lines = [line for line in lines[1:] if line.strip()]
    
    print(f'📊 Original rows: {len(data_lines)}')
    
    # Track seen match IDs
    seen = set()
    unique_lines = []
    
    for line in data_lines:
        match_id = line.split(',')[0]
        
        if match_id not in seen:
            seen.add(match_id)
            unique_lines.append(line)
    
    print(f'✅ Unique rows: {len(unique_lines)}')
    print(f'❌ Duplicates removed: {len(data_lines) - len(unique_lines)}')
    
    # Write cleaned CSV
    with open(output_file, 'w', encoding='utf-8') as f:
        f.write(header)
        f.writelines(unique_lines)
    
    print(f'💾 Saved to: {output_file}')

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print('Usage: python3 remove_csv_duplicates.py <input.csv> [output.csv]')
        sys.exit(1)
    
    input_file = sys.argv[1]
    output_file = sys.argv[2] if len(sys.argv) > 2 else None
    
    remove_duplicates(input_file, output_file)
